### PR TITLE
GitHub action to deploy docs on push to master

### LINKED ISSUE #7: Automated deployment of the documentation
Ideally docs would be deployed automatically on pushes to `master`. We should be able to make a GitHub action to do this.

### PR DESCRIPTION
**Pull request by birchmd**
I'm not 100% sure how to test whether this works or not. Probably we'll just have to merge this, then push some change to master and see if it runs successfully.

Closes #7

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Publish docs via GitHub Pages
```
Comment (artob) [inline .github/workflows/builds.yml:1]: Remember to begin YAML files with a `---` line. That will be caught by YAML lint tooling.

**Review by artob — APPROVED** [review]

File: .github/workflows/builds.yml (hunk 1 above)
Comment (artob) [reply .github/workflows/builds.yml:1]: Also, let's prefer `.yaml` over `.yml`.

Comment (artob) [reply .github/workflows/builds.yml:1]: Fixed in 506c9a6.

**Comment by artob** [comment]
@birchmd Works :+1: (Please note my comments for the future.)

- https://github.com/aurora-is-near/doc.aurora.dev/runs/2603381012
- https://github.com/aurora-is-near/doc.aurora.dev/commit/af4aef6438de09086c6d3b61b3d323ccc16f9830

**Comment by artob** [comment]
@birchmd On the downside, it does destroy all `gh-pages` branch history:

https://github.com/aurora-is-near/doc.aurora.dev/commits/gh-pages

Ideally that would be preserved so that we would know what exactly was published when. But it's not a high priority to fix.

**Comment by birchmd** [comment]
Agreed, that is not good. I assume this is because of [the `--force` option passed by the action](https://github.com/mhausenblas/mkdocs-deploy-gh-pages/blob/master/action.sh#L53).

The action is not very complex, so maybe we should copy the script locally to our repo and remove the `--force` option. Then use our own copy of the script in the action.

**Pull request merged**